pkg/utils/kubernetes/object.go

```diff
@@ -87,17 +88,57 @@ func DeleteObjectsFromListConditionally(ctx context.Context, c client.Client, li
 	return flow.Parallel(fns...)(ctx)
 }
 
-// ResourcesExist checks if there is at least one object of the given gvk. The kind in the gvk must be the list kind,
-// for example corev1.SchemeGroupVersion.WithKind("PodList").
-func ResourcesExist(ctx context.Context, reader client.Reader, gvk schema.GroupVersionKind, listOpts ...client.ListOption) (bool, error) {
-	objects := &metav1.PartialObjectMetadataList{}
-	objects.SetGroupVersionKind(gvk)
+// ResourcesExist checks if there is at least one object of the given objList.
+func ResourcesExist(ctx context.Context, reader client.Reader, objList client.ObjectList, scheme *runtime.Scheme, listOpts ...client.ListOption) (bool, error) {
+	objects := objList
+
+	// Use `PartialObjectMetadata` if no or metadata only field selectors are passed (informer's indexers only have access to metadata fields).
+	if hasNoOrMetadataOnlyFieldSelector(listOpts...) {
+		gvk, err := apiutil.GVKForObject(objList, scheme)
+		if err != nil {
+			return false, err
+		}
+
+		objects = &metav1.PartialObjectMetadataList{}
+		(objects.(*metav1.PartialObjectMetadataList)).SetGroupVersionKind(gvk)
```

Comment:
Looks a bit weird with parentheses at start.
```suggestion
		objects.(*metav1.PartialObjectMetadataList).SetGroupVersionKind(gvk)
```